File: 1 Algorithmic Toolbox/Week 3 Greedy Algorithms/week3_greedy_algorithms/1_money_change/change.py
# Uses python3
import sys

def get_change(m):
    #write your code here
    result = 0
    coins = [10, 5, 1]
    while m != 0:
        for coin in coins:
            result += m // coin
            m = m % coin
    
    return result

if __name__ == '__main__':
    m = int(sys.stdin.read())
    print(get_change(m))

File: 1 Algorithmic Toolbox/Week 3 Greedy Algorithms/week3_greedy_algorithms/1_money_change/change.cpp
#include <iostream>

int get_change(int m) {
  //write your code here
  int result = 0;
  int coins[3] = {10, 5, 1};
  
  while (m != 0){
      for (int i=0; i<3; i++){
          result += m / coins[i];
          m = m % coins[i];
          }
  }
  return result;
}

int main() {
  int m;
  std::cin >> m;
  std::cout << get_change(m) << '\n';
}
